src/transformers/trainer.py

```diff
@@ -2505,7 +2505,7 @@ def _inner_training_loop(
             for _ in range(total_updates):
                 update_step += 1
                 num_batches = args.gradient_accumulation_steps if update_step != (total_updates - 1) else remainder
-                batch_samples, num_items_in_batch = self.get_batch_samples(epoch_iterator, num_batches)
+                batch_samples, num_items_in_batch = self.get_batch_samples(epoch_iterator, num_batches, args.device)
```

Comment:
maybe we should use `accelerator.device` to get the right indexed device (`cuda:x`)

Reply:
or maybe it's not needed, just noticed args.device calls `_setup_devices` which calls accelerator